docs(fargate.md): init (#2963)
SumoLogic/sumologic-kubernetes-collection: docs/fargate.md

> # Fargate

Reviewer: Maybe let's add a "contents" section

Author: done below

export EFS_ID="$(
  aws efs describe-file-systems \
    --region="${AWS_REGION}" | \
    jq '.FileSystems[] |
      select(.Tags[].Key == "Name" and .Tags[].Value == "SumologicCollectionVolumes") |
      .FileSystemId' \
      --raw-output)"

if [[ -z "${EFS_ID}" ]]; then
  aws efs create-file-system \
    --encrypted \
    --performance-mode generalPurpose \
    --throughput-mode bursting \
    --tags Key=Name,Value=SumologicCollectionVolumes \
    --region "${AWS_REGION}"
  export EFS_ID="$(
    aws efs describe-file-systems \
      --region="${AWS_REGION}" | \
      jq '.FileSystems[] |
        select(.Tags[].Key == "Name" and .Tags[].Value == "SumologicCollectionVolumes") |
        .FileSystemId' \
        --raw-output)"
fi
```

`EFS_ID` is going to be used in setting up Volumes for metric pods.

## Metrics

You can install metrics with or without persistence. See the following sections for more details:

- [Persistence disabled](#persistence-disabled)
- [Persistence enabled](#persistence-enabled)

### Persistence Disabled

To disable persistence, add the following configuration to `user-values.yaml`:

```yaml
metadata:
  persistence:
    enabled: false
```

> **Note: This is going to disable persistence for logs and metrics.**

Reviewer: ```suggestion
**Note: This is going to disable persistence for both logs and metrics.**
```

### Persistence Enabled

> If you want to keep persistance (which is default configuration), you need to manually create Volumes for the Metric Pods. We recommend to

Reviewer: ```suggestion
If you want to keep persistence (which is default configuration), you need to manually create Volumes for the Metric Pods. We recommend to
```

> create Persistance Volume Claims on the top of EFS Storage. In order to set up them, please apply the following steps:

Reviewer: ```suggestion
create Persistent Volume Claims on the top of EFS Storage. In order to set up them, please apply the following steps:
```

> #### Create EFS access points for metric Pods

Author: done

  labels:
    app: ${HELM_INSTALLATION_NAME}-sumologic-otelcol-metrics
spec:
  capacity:
    storage: 10Gi
  volumeMode: Filesystem
  accessModes:
    - ReadWriteMany
  persistentVolumeReclaimPolicy: Retain
  storageClassName: efs-sc
  claimRef:
    namespace: ${NAMESPACE}
    name: file-storage-${HELM_INSTALLATION_NAME}-sumologic-otelcol-metrics-${counter}
  csi:
    driver: efs.csi.aws.com
    volumeHandle: ${EFS_ID}::${FSAP_ID}
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: file-storage-${HELM_INSTALLATION_NAME}-sumologic-otelcol-metrics-${counter}
  namespace: ${NAMESPACE}
  labels:
    app: ${HELM_INSTALLATION_NAME}-sumologic-otelcol-metrics
spec:
  accessModes:
    - ReadWriteMany
  storageClassName: efs-sc
  resources:
    requests:
      storage: 5Gi"
done | tee -a sumo-metrics-pvc.yaml
```

#### Create namespace if it doesn't exist already

```bash
## Create namespace if it doesn't exist already
kubectl create namespace "${NAMESPACE}"
```

#### Apply `sumo-metrics-pvc.yaml`

```bash
## Apply `sumo-metrics-pvc.yaml`
kubectl apply -f sumo-metrics-pvc.yaml -n "${NAMESPACE}"
```

### Install or upgrade collection

> Disable Prometheus Node Exporter. As Amazon Fargate does not support Daemonsets, they won't beworking correctly. There is no need to have

Reviewer: ```suggestion
Disable Prometheus Node Exporter. As Amazon Fargate does not support Daemonsets, they won't be working correctly. There is no need to have
```

    enabled: false
```

**Note: This is going to disable persistence for both logs and metrics.**

### Persistence Enabled

If you want to keep persistence (which is default configuration), you need to manually create Volumes for the Metric Pods. We recommend to
create Persistent Volume Claims on the top of EFS Storage. In order to set up them, please apply the following steps:

#### Create EFS access points for metric Pods

EFS Access Point is an entry point for an application. We recommend to create Acess Points pointing to the directories using
`/${NAMESPACE}/${PVC_NAME}` schema.

You can create them using the following `bash` script:

```bash
## Create EFS access points for metric Pods
for (( counter=0; counter<"${METRIC_PODS}"; counter++ )); do
  FSAP_ID="$(
    aws efs describe-access-points \
      --region "${AWS_REGION}" |
    jq ".AccessPoints[] |
      select(.RootDirectory.Path == \"/sumologic/file-storage-${HELM_INSTALLATION_NAME}-sumologic-otelcol-metrics-${counter}\") |
      .AccessPointId" \
      --raw-output)"

  if [[ -z "${FSAP_ID}" ]]; then
    aws efs create-access-point \
        --file-system-id "${EFS_ID}" \
        --posix-user Uid=1000,Gid=1000 \

>         --root-directory "Path=/${NAMESPACE}/file-storage-${HELM_INSTALLATION_NAME}-sumologic-otelcol-metrics-${counter},CreationInfo={OwnerUid=1000,OwnerGid=1000,Permissions=777}" \

Reviewer 2: don't we need to define access points for the paths below used in our helm charts?
- /conf
- /etc/otel
- /var/lib/storage/otc
- /tmp
- /etc/terraform
- /customer-scripts

    name: file-storage-${HELM_INSTALLATION_NAME}-sumologic-otelcol-metrics-${counter}
  csi:
    driver: efs.csi.aws.com
    volumeHandle: ${EFS_ID}::${FSAP_ID}
---
apiVersion: v1
kind: PersistentVolumeClaim
metadata:
  name: file-storage-${HELM_INSTALLATION_NAME}-sumologic-otelcol-metrics-${counter}
  namespace: ${NAMESPACE}
  labels:
    app: ${HELM_INSTALLATION_NAME}-sumologic-otelcol-metrics
spec:
  accessModes:
    - ReadWriteMany
  storageClassName: efs-sc
  resources:
    requests:
      storage: 5Gi"
done | tee -a sumo-metrics-pvc.yaml
```

#### Create namespace if it doesn't exist already

```bash
## Create namespace if it doesn't exist already
kubectl create namespace "${NAMESPACE}"
```

#### Apply `sumo-metrics-pvc.yaml`

```bash
## Apply `sumo-metrics-pvc.yaml`
kubectl apply -f sumo-metrics-pvc.yaml -n "${NAMESPACE}"
```

### Install or upgrade collection

Disable Prometheus Node Exporter. As Amazon Fargate does not support Daemonsets, they won't be working correctly. There is no need to have
Node Exporter on Amazon Fargate as Amazon takes care of Nodes management. To disable Prometheus Node Exporter, please add the following
configuration to `user-values.yaml`:

> ```yaml

Reviewer 2: I think we also need this in our use-values.yaml:

```
sumologic:
  collector:
    logs:
      collector:
        otelcol:
          enabled: false
```

Author: yeah, but this specific section is about metrics signal

Author: If we don't want to merge unfinished document we can work on this branch, but lets keep it consistent, so refactor it to write about all signals the same time or split it into sections. Now it's splited, but maybe it would be better to slightly reorganize it

Author: any of the mentioned change should be subject of separate PR in my opinion
